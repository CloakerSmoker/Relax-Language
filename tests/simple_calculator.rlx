#Require ./src/lib/String.rlx
#Require ./src/lib/Console.rlx

/* 
	A simple calculator, which works when compiled for both Windows and Linux.
*/

#ImageBase 0x30000000

/*
ARGS: 1 + 2 29 - 10 8 "*" 5 60 / 3
*/

define i32 Main(i64 ArgC, i8** ArgV) {
	GetArgs(&ArgC, &ArgV) /* Ensures that ArgC/ArgV are set on Windows, does nothing on Linux */
	
	i64 Count := (ArgC - 1) / 3

	for (i64 Index := 0, Index < Count, Index++) {
		i64 Start := (Index * 3) + 1

		i64 Left := AToI(ArgV[Start + 0])
		i64 Right := AToI(ArgV[Start + 2])
		
		i8 Operator := ArgV[Start + 1][0]

		if (Operator = '+') {
			Print(Left + Right)
		}
		else if (Operator = '-') {
			Print(Left - Right)
		}
		else if (Operator = '*') {
			Print(Left * Right)
		}
		else if (Operator = '/') {
			Print(Left / Right)
		}

		Print(" ")
	}
	
	return 0 /* Returns from `Main()` like normal on Windows, calls `sys_exit(0)` on Linux */
}
