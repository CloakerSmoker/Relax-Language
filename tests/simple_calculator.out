3 19 40 20 